deps/thpool/thpool.c

```diff
@@ -175,16 +271,15 @@ struct redisearch_thpool_t* redisearch_thpool_create(size_t num_threads) {
   return thpool_p;
 }
 
-/* Initialise thread pool */
-void redisearch_thpool_init(struct redisearch_thpool_t* thpool_p, LogFunc log) {
-  assert(thpool_p->keepalive == 0);
-  thpool_p->keepalive = 1;
-  thpool_p->terminate_when_empty = 0;
-
+/* initialize thread pool */
+void redisearch_thpool_init(struct redisearch_thpool_t* thpool_p) {
+  assert(!(thpool_p->flags & RS_THPOOL_F_KEEP_ALIVE));
+  redisearch_thpool_TURNON_flag(thpool_p, RS_THPOOL_F_KEEP_ALIVE);
+  redisearch_thpool_TURNOFF_flag(thpool_p, RS_THPOOL_F_TERMINATE_WHEN_EMPTY);
```

Comment:
Why we need the `redisearch_` prefix? I look at thpool.c as a general thread pool implementation so we should avoid module specific names.

Reply:
@MeirShpilraien This is due to use of RediSearch within RedisGraph, to avoid namespace clashes.